src/dac.rs

```diff
@@ -0,0 +1,233 @@
+//! Configure the internal DAC on the stm32f3xx.
+//! Incomplete, but includes basic operation.
+
+use core::fmt;
+
+use crate::{
+    gpio::{
+        gpioa::{PA4, PA5},
+        Analog,
+    },
+    pac,
+    rcc::APB1,
+};
+
+/// Trait representing a single-channel digital-to-analog converter (DAC).
+pub trait SingleChannelDac<Word> {
+    /// Error type returned by DAC methods
+    type Error;
+
+    /// Output a constant signal, given a bit word.
+    fn try_set_value(&mut self, value: Word) -> Result<(), Self::Error>;
+}
+
+/// This is an abstraction to ensure that the DAC output pin is configured
+/// as an analog output.
+pub trait Pins {}
+impl Pins for PA4<Analog> {}
+impl Pins for PA5<Analog> {}
+
+#[derive(Clone, Copy, Debug)]
+/// Select the channel
+pub enum Channel {
+    /// Channel 1
+    One,
+    /// Channel 2
+    Two,
+}
+
+#[derive(Clone, Copy, Debug)]
+/// Three options are available to set DAC precision.
+pub enum DacBits {
+    /// Eight bit precision, right-aligned.
+    EightR,
+    /// 12-bit precision, left-aligned.
+    TwelveL,
+    /// 12-bit precision, right-aligned.
+    TwelveR,
+}
+
+#[derive(Clone, Copy)]
+/// Select a trigger, used by some features.
+pub enum Trigger {
+    /// Timer 6
+    Tim6,
+    /// Timers 3 or 8
+    Tim3_8,
+    /// Timer 7
+    Tim7,
+    /// Timer 15
+    Tim15,
+    /// Timger 2
+    Tim2,
+    /// Tim4 is Not available on DAC 2.
```

Comment:
Since we don't support DAC2 anymore, the "not available" not here is not necessary either